cscs-checks/apps/vasp/vasp_check.py

```diff
@@ -7,105 +7,121 @@
-    }
+        self.skip_if(self.num_nodes not in valid_systems,
+                     f'No valid systems found for {self.num_nodes}(s)')
+        self.valid_systems = valid_systems[self.num_nodes]
 
-    @run_after('init')
-    def setup_by_scale(self):
-        self.descr = f'VASP CPU check'
-        if self.current_system.name == 'dom':
-            self.num_tasks = 72
-            self.num_tasks_per_node = 12
-            self.use_multithreading = True
-        elif self.current_system.name in ['eiger', 'pilatus']:
-            self.num_tasks = 64
-            self.num_tasks_per_node = 4
-            self.num_cpus_per_task = 8
-            self.num_tasks_per_core = 1
-            self.use_multithreading = False
-            self.variables = {
-                'MPICH_OFI_STARTUP_CONNECT': '1',
-                'OMP_NUM_THREADS': str(self.num_cpus_per_task),
-                'OMP_PLACES': 'cores',
-                'OMP_PROC_BIND': 'close'
-            }
+        # setup programming environment filter
+        if self.current_system.name in ['eiger', 'pilatus']:
+            self.valid_prog_environs = ['cpeIntel']
         else:
-            self.num_tasks = 32
-            self.num_tasks_per_node = 2
-            self.use_multithreading = True
-
-        self.reference = self.references_by_variant[self.variant]
-        self.tags |= {'maintenance', 'production'}
+            self.valid_prog_environs = ['builtin']
 
-    @run_before('run')
-    def set_task_distribution(self):
-        self.job.options = ['--distribution=block:block']
 
     @run_before('run')
-    def set_cpu_binding(self):
+    def setup_run(self):
+        # set auto-detected architecture
+        self.skip_if_no_procinfo()
+        proc = self.current_partition.processor
+        arch = proc.arch
+
+        # set architecture for GPU partition (no auto-detection)
+        if self.current_partition.fullname in ('daint:gpu', 'dom:gpu'):
+            arch = 'sm_60'
+
+        try:
+            found = self.references[self.num_nodes][arch]
+        except KeyError:
+            self.skip(f'Configuration with {self.num_nodes} node(s) '
+                      f'is not supported on {arch!r}')
+
+        # common setup for every architecture
         self.job.launcher.options = ['--cpu-bind=cores']
+        self.job.options = ['--distribution=block:block']
+        self.num_tasks_per_node = proc.num_sockets
+        self.num_cpus_per_task = proc.num_cores // self.num_tasks_per_node
+        self.num_tasks = self.num_nodes * self.num_tasks_per_node
+        self.variables = {
+            'OMP_NUM_THREADS': str(self.num_cpus_per_task),
+            'OMP_PLACES': 'cores',
+            'OMP_PROC_BIND': 'close'
+        }
 
+        # custom settings for selected architectures
+        if arch == 'sm_60':
+            self.num_gpus_per_node = 1
+        elif arch == 'zen2':
+            self.variables += {
+                'MPICH_OFI_STARTUP_CONNECT': '1'
+            }
```

Comment:
```suggestion
            self.variables.update({
                'MPICH_OFI_STARTUP_CONNECT': '1'
            })
```

The `+=` does not work on dictionaries. This path has not been executed by the CI, because it is temporarily disabled for Eiger/Pilatus.

Reply:
Thanks for pointing it out!